vega/utils.py

```diff
@@ -43,9 +44,11 @@ def sanitize_dataframe(df):
     if isinstance(df.columns, pd.core.index.MultiIndex):
         raise ValueError('Hierarchical indices not supported')
 
-    def to_list_if_array(val):
+    def parse_object_column_type(val):
         if isinstance(val, np.ndarray):
             return val.tolist()
+        elif isinstance(val, datetime.date):
+            return "{dt:%Y-%m-%d}".format(dt=val)
```

Comment:
Date parsing has to be done very carefully in Vega-Lite: these dates should be encoded in ISO format, or else you will run into bugs like this one: https://github.com/altair-viz/altair/issues/1027

Reply:
I completely agree that having numpy types is strongly preferred over python date types. Although I don't think it is unreasonable to assume that a column of python date be serialized into valid json for plotting. 

Adding this support will benefit anyone converting Apache Spark DataFrame to Pandas DataFrame because DateType columns are cast to python dates. You can see in [SPARK-23290](https://issues.apache.org/jira/browse/SPARK-23290) the reasoning for this design choice.

I did add an explicit cast to iso format